textpipe/doc.py

```diff
@@ -509,3 +510,46 @@ def aggregate_word_vectors(self, model_name=None, aggregation='mean', normalize=
             return numpy.var(vectors, axis=0).tolist()
         else:
             raise NotImplementedError(f'Aggregation method {aggregation} is not implemented.')
+
+    def _load_gensim_word2vec_model(self, model_file=None):
+        """
+        Loads pre-trained Gensim word2vec model, expects lower-cased training data
+        >>> model = Doc('')._load_gensim_word2vec_model('tests/models/gensim_test_nl.w2v')
+        >>> type(model)
+        <class 'gensim.models.keyedvectors.Word2VecKeyedVectors'>
+        """
+        lang = self.language if self.is_reliable_language else self.hint_language
+        if not self._gensim_vectors or lang not in self._gensim_vectors:
+            vectors = Word2Vec.load(model_file).wv
+            self._gensim_vectors[lang] = vectors
+        return self._gensim_vectors[lang]
+
+    @functools.lru_cache()
+    def generate_gensim_document_embedding(self, model_file=None):
```

Comment:
Not sure if it's in the scope of this task but IMHO it would be nice to mirror the existing `word_vectors`, `generate_word_vectors`, `aggregate_word_vectors`, `doc_vector`-functionality we have for spacy vectors (i.e., also allow us the yield a list of Gensim word vectors, and using different doc/aggregation strategies.

Reply:
We initially tried it this way indeed, but got stuck with the provided model_mapping that becomes ambiguous in aggregate_word_vectors(), since this function would then require a spacy model mapping and a word2vec model mapping :/

Reply:
Hmm, yeah, but maybe we can address that by passing as a `kwarg` the model/type to pick? Anyway, maybe better for an incremental improvement/refinement later down the line.

Reply:
(unless other reviewers feel strongly about this, cc @anneschuth @dodijk)

Reply:
When there's a benefit of merging the two models/types than I'd be in favour of consolidating, until that's not the case I'm fine with keeping it apart.

Reply:
OK, but other than consolidating vs. keeping apart, what about mirroring functionality (i.e., at least also supply `word_vector` functionality?).